src/calculation/DetailedResultsBuilder.ts

```diff
@@ -52,12 +52,32 @@ export function createPopulationValues(
       // create patient level population and stratifier results based on episodes
       episodeResults.forEach(episodeResult => {
         episodeResult.populationResults.forEach(popResult => {
-          createOrSetResult(
-            popResult.populationType,
-            popResult.criteriaExpression,
-            popResult.result,
-            populationResults
+          const measureObsPops = populationResults.find(
+            pop => pop.populationType == PopulationType.OBSERV && popResult.criteriaExpression == pop.criteriaExpression
           );
+          if (!measureObsPops) {
+            // only add observations if it is not undefined
+            if (popResult.observations?.[0]) {
+              createOrSetResult(
+                popResult.populationType,
+                popResult.criteriaExpression,
+                popResult.result,
+                populationResults,
+                popResult.observations as string[]
+              );
+            } else {
+              createOrSetResult(
+                popResult.populationType,
+                popResult.criteriaExpression,
+                popResult.result,
+                populationResults
+              );
+            }
+          } else {
```

Comment:
Nat and I found an issue with this branch of logic that affects multiple IPP support and a few other things. Just noting it here, but we'll discuss in a call